src/app/commands/cmd_new_layer.cpp

```diff
@@ -206,6 +207,10 @@ void NewLayerCommand::onExecute(Context* context)
   tilesetInfo.baseIndex = 1;
 
 #ifdef ENABLE_UI
+  auto& pref = Preferences::instance();
+  tilesetInfo.grid = doc::Grid(gfx::Size(pref.tileset.gridWidth(),
+                                         pref.tileset.gridHeight()));
```

Comment:
I think we shouldn't save the grid width/height in preferences, as the current grid settings are used anyway (see `tilesetInfo.grid = context->activeSite().grid();` above) and I think we prefer that one as the default.